tangelo/toolboxes/molecular_computation/rdms.py

```diff
@@ -82,11 +81,16 @@ def compute_rdms(ferm_ham, mapping, up_then_down, exp_vals=None, exp_data=None,
         raise RuntimeError("Arguments exp_vals, exp_data and shadow are mutually exclusive. Provide exactly one of them.")
 
     # Initialize exp_vals
-    exp_vals = {pauli_string_to_of(term): exp_val for term, exp_val in exp_data.items()} \
-        if isinstance(exp_vals, dict) else {}
+    if isinstance(exp_vals, dict) and set(map(type, exp_vals)) == {str}:
```

Comment:
A beautiful and compact way to use map to check all the keys in a dictionary for a given type. Thank you !